### PR TITLE
Update npm token for publishing

### PR DESCRIPTION
**Pull request by bkjohnson**
## Description
Closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/36875

This updates the CI job for publishing on a release to use an npm token provided by ops instead of my personal one.

## Testing done

- Checked to see if the token signed us in as the bot.

I don't want to publish anything just as a test run, and I didn't find a `--dry-run` or anything similar for publishing. We should find out if this works if this job fails on our next release.

## Screenshots

Signing in as the right npm account using the provided auth token:

bot account name highlighed as `npm whoami` output

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

**Review by bkjohnson — COMMENTED** [review]
The bot is now part of our npm org, so it should have the right permissions for publishing. We'll find out the next time we make a release :smile: 

image

Thank you @oseasmoran73 for the help!

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -14,6 +14,18 @@ jobs:
       with:
         node-version: 14.x
         cache: 'yarn'
+    - uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: "us-gov-west-1"
```
Comment (bkjohnson) [inline .github/workflows/publish.yml:21]: We're doing the same thing in our `publish-storybook` workflow, so I copied that:

https://github.com/department-of-veterans-affairs/component-library/blob/e157e3c011afb64c8431cc7c7b266ab4c786ac03/.github/workflows/publish-storybook.yml#L34-L38

**Review by oseasmoran73 — APPROVED** [review]
LGTM. Thanks for being patient with me @bkjohnson !

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -14,6 +14,18 @@ jobs:
...
+
+    - name: Get bot token from Parameter Store
+      uses: marvinpinto/action-inject-ssm-secrets@v1.2.1
+      with:
+        ssm_parameter: /devops/VA_VSP_BOT_NPM_TOKEN
+        env_variable_name: VA_VSP_BOT_NPM_TOKEN
```
Comment (bkjohnson) [inline .github/workflows/publish.yml:27]: This pattern is copied from elsewhere in [`content-build`](https://github.com/department-of-veterans-affairs/content-build/blob/5f6b464ece2b595b3bd9af29c1b773bda95700e4/.github/workflows/continuous-integration.yml#L252-L265)

**Review by pdavies88 — APPROVED** [review]

**Pull request merged**